web/client/components/TOC/fragments/settings/Display.jsx

```diff
@@ -28,40 +29,78 @@ module.exports = class extends React.Component {
+    };
+
+    componentDidMount() {
+        if (this.props.settings && this.props.settings.options && !isNil(this.props.settings.options.opacity)) {
+            this.setState({ opacity: Math.round(this.props.settings.options.opacity * 100)});
+        }
+    }
+
     render() {
-        return (<Grid fluid style={{paddingTop: 15, paddingBottom: 15}}>
-            {this.props.element.type === "wms" ?
-            [(<label key="format-label" className="control-label"><Message msgId="layerProperties.format" /></label>),
-            (<DropdownList
-                key="format-dropdown"
-                data={this.props.formats || ["image/png", "image/png8", "image/jpeg", "image/vnd.jpeg-png", "image/gif"]}
-                value={this.props.element && this.props.element.format || "image/png"}
-                onChange={(value) => {
-                    this.props.onChange("format", value);
-                }} />)] : null}
-            <div key="opacity">
-            <label key="opacity-label" className="control-label">{this.props.opacityText}</label>
-            <Slider key="opacity-slider" start={[Math.round(this.props.settings.options.opacity * 100)]}
-                range={{min: 0, max: 100}}
-                onChange={(opacity) => {this.props.onChange("opacity", opacity / 100); }}/>
-            <Label key="opacity-percent" >{Math.round(this.props.settings.options.opacity * 100) + "%"}</Label>
-            </div>
-            {this.props.element.type === "wms" ?
-                [(<div key="transparent-check">
-                    <Checkbox key="transparent" checked={this.props.element && (this.props.element.transparent === undefined ? true : this.props.element.transparent)} onChange={(event) => {this.props.onChange("transparent", event.target.checked); }}>
-                        <Message msgId="layerProperties.transparent"/></Checkbox>
-                    <Checkbox key="cache" value="tiled" key="tiled"
-                        disabled={!!this.props.element.singleTile}
-                        onChange={(e) => this.props.onChange("tiled", e.target.checked)}
-                        checked={this.props.element && this.props.element.tiled !== undefined ? this.props.element.tiled : true} >
-                        <Message msgId="layerProperties.cached"/>
-                    </Checkbox>
-                    <Checkbox key="singleTile" value="singleTile" key="singleTile"
-                        checked={this.props.element && (this.props.element.singleTile !== undefined ? this.props.element.singleTile : false )}
-                        onChange={(e) => this.props.onChange("singleTile", e.target.checked)}>
-                        <Message msgId="layerProperties.singleTile"/>
-                    </Checkbox>
-                </div>)] : null}
-        </Grid>);
+        return (
+            <Grid
+                fluid
+                style={{paddingTop: 15, paddingBottom: 15}}>
+                {this.props.element.type === "wms" &&
```

Comment:
This may render false on no wms layer, did you tested it with WMTS layers?

Reply:
comment below